src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -325,61 +325,6 @@ impl LogicalAggBuilder {
         None
     }
 
-    /// syntax check for distinct aggregates.
```

Comment:
By removing this `syntax_check`, any agg call combination that cannot be rewritten by `DistinctAggRule` will fallback to backend implementation.